mirror.py

```diff
@@ -20,14 +20,14 @@ def main():
     for version in target_versions:
         paths = process_version(version)
         if subprocess.check_output(["git", "status", "-s"]).strip():
-            subprocess.run(["git", "add", *paths])
-            subprocess.run(["git", "commit", "-m", f"Mirror: {version}"])
-            subprocess.run(["git", "tag", f"v{version}"])
+            subprocess.run(["git", "add", *paths], check=True)
+            subprocess.run(["git", "commit", "-m", f"Mirror: {version}"], check=True)
+            subprocess.run(["git", "tag", f"v{version}"], check=True)
         else:
             print(f"No change v{version}")
 
 
-def get_all_versions() -> typing.List[Version]:
+def get_all_versions() -> list[Version]:
```

Comment:
Do we need `from __future__ import annotations` or is it clear that this script requires a modern Python version?

Reply:
I just double-checked, and this is only relevant for development. I was still able to run this hook no problem in a python 3.8 environment. So I personally think it is fine to make this assumption. The only thing that matters, is that the workflow python version is modern enough (and it is):
https://github.com/astral-sh/ruff-pre-commit/blob/6e85b5cd1f7a66f3a749fa23c513546a3b649f34/.github/workflows/main.yml#L19-L21

Reply:
We're importing `tomllib` at the top of the file anyway (and were doing so prior to this PR), so I think this script will only run on py311+